google-cloud-firestore/src/main/java/com/google/cloud/firestore/FieldMask.java

```diff
@@ -62,16 +60,18 @@ public static FieldMask of(String... fieldPaths) {
    */
   @Nonnull
   public static FieldMask of(FieldPath... fieldPaths) {
-    return new FieldMask(Arrays.asList(fieldPaths));
+    TreeSet<FieldPath> paths = new TreeSet<>();
+    Collections.addAll(paths, fieldPaths);
+    return new FieldMask(paths);
   }
 
   static FieldMask fromObject(Map<String, Object> values) {
-    List<FieldPath> fieldPaths = extractFromMap(values, FieldPath.empty());
+    TreeSet<FieldPath> fieldPaths = extractFromMap(values, FieldPath.empty());
     return new FieldMask(fieldPaths);
   }
 
-  private static List<FieldPath> extractFromMap(Map<String, Object> values, FieldPath path) {
-    List<FieldPath> fieldPaths = new ArrayList<>();
+  private static TreeSet<FieldPath> extractFromMap(Map<String, Object> values, FieldPath path) {
```

Comment:
A potentially better optimization here would be to modify the `extractFromMap()` method to take the destination collection as an argument, instead of creating a new collection on every invocation. This is especially true since this method calls itself recursively and creates a new collection on each invocation.

Reply:
Nevermind. @tom-andersen pointed out to me that TreeSet was already there; this change avoids creating an intermediary collection that is (almost) immediately converted to a TreeSet then thrown away.